sys/net/application_layer/nanocoap/Makefile.include

```diff
@@ -1,4 +1,8 @@
 ifneq (,$(filter nanocoap_dtls,$(USEMODULE)))
+  # join will sort the list, subst selects the 2nd element
+  max = $(subst xx,x,$(join $1,$2))
   CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT := $(or $(CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT),2)
-  DTLS_MAX_BUF ?= ((1 << ($(CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT) + 3)) + 36)
+  # the initial DTLS handshake may exceed the block size
+  NANOCOAP_BLOCK_FRAME := $((1 << ($(CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT) + 3)) + 36
```

Comment:
Does this kind of calculation work for you?

```makefile

  NANOCOAP_BLOCK_FRAME := $((1 << ($(CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT) + 3)) + 36)
  $(info NANOCOAP_BLOCK_FRAME=$(NANOCOAP_BLOCK_FRAME))
  DTLS_MAX_BUF ?= $(call max,$(NANOCOAP_BLOCK_FRAME),200)
  $(info DTLS_MAX_BUF=$(DTLS_MAX_BUF))
```

prints

```
fabian.huessler@ml-pa.loc@MLPA-NB119:~/RIOT$ USEMODULE+="gnrc_icmpv6_echo nanocoap_dtls" BOARD=same54-xpro make -C tests/net/nanocoap_cli flash 
make: Entering directory '/home/fabian.huessler@ml-pa.loc/RIOT/tests/net/nanocoap_cli'
NANOCOAP_BLOCK_FRAME=
DTLS_MAX_BUF=200
```

And this:
```makefile
  NANOCOAP_BLOCK_FRAME := $(shell echo $$(( ((1 << ($(CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT) + 3)) + 36) )))
  $(info NANOCOAP_BLOCK_FRAME=$(NANOCOAP_BLOCK_FRAME))
  DTLS_MAX_BUF ?= $(call max,$(NANOCOAP_BLOCK_FRAME),200)
  $(info DTLS_MAX_BUF=$(DTLS_MAX_BUF))
```

prints:

```
fabian.huessler@ml-pa.loc@MLPA-NB119:~/RIOT$ USEMODULE+="gnrc_icmpv6_echo nanocoap_dtls" BOARD=same54-xpro make -C tests/net/nanocoap_cli flash 
make: Entering directory '/home/fabian.huessler@ml-pa.loc/RIOT/tests/net/nanocoap_cli'
NANOCOAP_BLOCK_FRAME=68
DTLS_MAX_BUF=68200
```